rootfs/api/models/certificate.py

```diff
@@ -220,9 +221,8 @@ def detach(self, *args, **kwargs):
                 # We raise an exception when a secret doesn't exist
                 self._scheduler._get_secret(app, name)
                 self._scheduler._delete_secret(app, name)
-            except KubeHTTPException as e:
-                logger.critical(e)
-                raise EnvironmentError("Could not delete certificate secret {} for application {}".format(name, app))  # noqa
+            except KubeException as e:
+                raise ServiceUnavailable("Could not delete certificate secret {} for application {}".format(name, app)) from e  # noqa
```

Comment:
It doesn't when service unavailable, it has the error message. I think you are just looking at the default message